Added Supabase example using only javascript.

preetsuthar17 opened the pull request:
I made simple example of using Supabase in normal JavaScript which basically takes input from user and saves it in database.

preetsuthar17 committed:
Added Supabase example using only javascript.

Isaiah-Hamilton commented:
Kind of confused what the example exactly does.  can you create a `README.md` for what this example does/what its about and how to set it up.

preetsuthar17 committed:
Create README.md

preetsuthar17 committed:
Merge branch 'supabase:master' into patch-1

preetsuthar17 commented:
> Kind of confused what the example exactly does. can you create a `README.md` for what this example does/what its about and how to set it up.

Yeahh done! i forgot to select that file earlier, my mistake.

gregnr commented on examples/simple-username-database/README.md:
```diff
@@ -0,0 +1,71 @@
+
+# Supabase Project Documentation
+
+This documentation provides an overview of a JavaScript project that utilizes Supabase for the database. The project allows users to enter their username.It is an open-source project created by [preetsuthar17](https://github.com/preetsuthar17) and can be found on [GitHub](https://github.com/preetsuthar17).
```
@preetsuthar17 ideally if we're bringing this example to Supabase we should keep the links relative to this example.

```suggestion
This documentation provides an overview of a JavaScript project that utilizes Supabase for the database. The project allows users to enter their username.
```

gregnr commented on examples/simple-username-database/README.md:
````diff
@@ -0,0 +1,71 @@
+
+1. Clone the repository from GitHub:
+   ```
+   git clone https://github.com/preetsuthar17/supabase-gemo.git
````
Here as well 👆

preetsuthar17 replied:
After merging pull request we can change that link to this repository.

gregnr commented on examples/simple-username-database/README.md:
````diff
@@ -0,0 +1,71 @@
+
+Once the project is set up, you can run it using the following command:
+
+```
+node index.js
+```
````
```suggestion
```shell
node index.js
```
```

gregnr commented on examples/simple-username-database/README.md:
````diff
@@ -0,0 +1,71 @@
+node index.js
+```
+
+[![Demo Vid]()](https://github.com/preetsuthar17/supabase-demo/assets/75468116/b5e88f63-525c-4211-8c50-b46e198d4632)
+
+
+![image](https://github.com/preetsuthar17/supabase-demo/assets/75468116/12f5a4e2-8a42-4b65-80d5-ede9161d6c0c)
````
We should upload these to this repo if we plan to include them.

preetsuthar17 replied:
Alright, I'll add the image and video to the repository.

gregnr commented on examples/simple-username-database/index.js:
```diff
@@ -0,0 +1,70 @@
+
+// one of the most simplest example of javascript using supabase datbase to take input from user and save it in database.
```
```suggestion
// Simple JavaScript example to take a username input and save it in a Supabase database.
```

gregnr commented on examples/simple-username-database/index.js:
```diff
@@ -0,0 +1,70 @@
+const { createClient } = require("@supabase/supabase-js");
+const readline = require("readline");
+
+const supabaseUrl = "replace-with-your-supabase-url";
+const supabaseKey =
+  "replace-with-your-supabasekey";
+const supabase = createClient(supabaseUrl, supabaseKey);
```
We can probably combine these into one line for simplicity

gregnr commented on examples/simple-username-database/index.js:
```diff
@@ -0,0 +1,70 @@
+
+// one of the most simplest example of javascript using supabase datbase to take input from user and save it in database.
+
+const { createClient } = require("@supabase/supabase-js");
+const readline = require("readline");
```
Should we update to modern ES Module imports?

gregnr replied:
@preetsuthar17 thoughts on these other comments? I see we are still using `require()`.

gregnr commented on examples/simple-username-database/index.js:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    console.log("Inserted name:", name);
+    import("chalk").then((chalk) => {
```
Ideally we follow best practices by importing dependencies at the top.

gregnr commented on examples/simple-username-database/package.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "supabase-test",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@supabase/supabase-js": "^2.26.0",
+    "chalk": "^5.2.0",
+    "readline": "^1.3.0"
+  }
+}
```
Let's update these meta fields to match what we do in our other examples

gregnr reviewed (commented):
Thanks for the PR @preetsuthar17. Left some comments/feedback for you.

preetsuthar17 committed:
Update examples/simple-username-database/README.md

Co-authored-by: Greg Richardson <greg.nmr@gmail.com>

preetsuthar17 committed:
Update examples/simple-username-database/README.md

Co-authored-by: Greg Richardson <greg.nmr@gmail.com>

preetsuthar17 committed:
Update examples/simple-username-database/index.js

Co-authored-by: Greg Richardson <greg.nmr@gmail.com>

preetsuthar17 commented:
It's all done! please check and if anything needs to be changed let me know. Some dependencies are just not working by including them on top.

preetsuthar17 committed:
Add files via upload

preetsuthar17 committed:
Delete Screenshot 2023-06-25 141440.png

preetsuthar17 committed:
Add files via upload

preetsuthar17 committed:
Update README.md

preetsuthar17 committed:
Add files via upload

preetsuthar17 committed:
Update README.md

gregnr commented:
Thanks for the updates @preetsuthar17. Taking a second look through this example - I think there's a few more things we should address:
- We should be clear (in the title / readme description) that this is a way to connect to Supabase from Node.js. I think indicating Node.js is key so that developers are clear on what it's intended for
- I'm hesitant to use 'usernames' as the table in this example, since usernames are usually tied closely with authentication which we don't include here. Could we come up with another entity that might be less confusing/less involved for developers?
- We should include the table schema in the README/migrations file so that developers can reproduce

preetsuthar17 commented:
can you suggest me name for the table?

preetsuthar17 committed:
Update README.md

preetsuthar17 committed:
Add files via upload

preetsuthar17 committed:
Update README.md

preetsuthar17 committed:
Username replaced with name-reg

preetsuthar17 committed:
change name of project

preetsuthar17 commented:
I did minor changes in readme files and other files by changing the project name from username to name-reg. updated the readme and added the guide to generate the table in the dashboard using the SQL editor query.

gregnr commented:
Thanks @preetsuthar17 - maybe instead of names we can store something like "todos"?

preetsuthar17 commented:
That would be way better than names, I will change this for sure! i am busy for a while so I'll do it later.

preetsuthar17 committed:
changed project from name-reg to simple 
To-Do

preetsuthar17 commented:
@gregnr I changed the project name with it functionality tot todo

preetsuthar17 committed:
update codee

preetsuthar17 committed:
change SQL Query

preetsuthar17 commented:
@gregnr All done! I added functionality to fetch data from the database and add data to the database via CMD. I also updated the table's name to `todoreg` and the rows in there!

preetsuthar17 commented:
@gregnr you here?

gregnr commented:
Thanks @preetsuthar17. By chance did you see my comments earlier w.r.t. using `import` statements instead of `require()`? Also had another one about the meta fields in `package.json`. Let me know if you have any questions about those.

preetsuthar17 committed:
Modern ES imports, Update package.json

preetsuthar17 commented:
@gregnr I'm really sorry I have made updates to the imports and package.json file Please tell me if there is anything else that needs to be changed.

preetsuthar17 commented:
@gregnr is there anything else left to change?
